src/sardana/taurus/core/tango/sardana/test/test_macro.py

```diff
@@ -114,3 +121,175 @@ def verifyXML(self, macro_name, param_def, param_value, expected_xml_rep):
+#            macro_params=[[[]]],
+#            expected_params_list=[[['mot01', "50"]]],
+#            )
+
+#### Testing list of elements (moveables) followed by a ####
+################# single parameter (float). ################
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt10d_param_def,
+            macro_params=[["1", "3", "15"], "mot01"],
+            expected_params_list=[["1", "3", "15"], "mot01"]
+            )
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt10d_param_def,
+            macro_params=[["1", "3", []], "mot01"],
+            expected_params_list=[["1", "3", "100"], "mot01"]
+            )
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt10d_param_def,
+            macro_params=[["1", "3", "15"], []],
+            expected_params_list=[["1", "3", "15"], "mot99"]
+            )
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt10d_param_def,
+            macro_params=[["1", [], "15"], []],
+            expected_params_list=[["1", "100", "15"], "mot99"],
+            )
+
+#### Testing lists of elements (floats), followed by another list of ####
+######################### elements (moveables). #########################
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt12d_param_def,
+            macro_params=[["1", "3", "4"], ["mot1", "mot2"]],
+            expected_params_list=[["1", "3", "4"], ["mot1", "mot2"]]
+            )
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt12d_param_def,
+            macro_params=[["1", [], "4"], [[], "mot2"]],
+            expected_params_list=[["1", "100", "4"], ["mot99", "mot2"]]
+            )
+
+########################### Testing nested paramRepeats ########################
+############## Groups of motors, where each group is a motor list ##############
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt13d_param_def,
+            macro_params=[[["mot2", "mot3"], ["mot4", "mot5", "mot6"]]],
+            expected_params_list=[[["mot2", "mot3"], ["mot4", "mot5", "mot6"]]]
+            )
+
+########################### Testing nested paramRepeats ########################
+## Groups of motors, where each group is a motor list followed by a number #####
+@insertTest(helper_name='verifyEncoding',
+            param_def=pt14d_param_def,
+            macro_params=[[[["mot2", "mot3"], 4],
+                           [["mot4", "mot5", "mot6"], 5]]],
+            expected_params_list=[[[["mot2", "mot3"], 4],
+                                   [["mot4", "mot5", "mot6"], 5]]]
+            )
+
+class ParamsTestCase(unittest.TestCase):
+
+    def verifyEncoding(self, param_def, macro_params, expected_params_list):
+        """
+        Helper to verify the correct building of the parameters objects tree.
+        Verify that the list is recreated correctly from the parameters
+        objects tree.
+        :param macro_name: (str) name of the macro
```

Comment:
The comment "param macro_name: (str) name of the macro" should be removed because it is not used anymore.